go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -2000,7 +2000,7 @@ func (tsv *TabletServer) SetPoolSize(val int) {
 	if val <= 0 {
 		return
 	}
-	tsv.qe.conns.SetCapacity(int64(val))
+	_ = tsv.qe.conns.SetCapacity(context.Background(), int64(val))
```

Comment:
as no timeout is set here. `capacity` mutex can be held forever and pool close would remain blocked.

there are 2 other places like that here.